File: episode-16-octave/fib.m
#!/usr/bin/env octave

u = [1,1];
m = [1 1; 1 0];

for i = 1:40
  result = dot(u * (m**(i-1)), [0 1]);
  printf("fib(%d) = %d\n", i, result);
endfor

File: episode-100-ruby/fib.rb
#!/usr/bin/env ruby

def fib(n)
  if n <= 2
    1
  else
    fib(n-1) + fib(n-2)
  end
end

(1..30).each do |n|
  puts "fib(#{n}) = #{fib(n)}"
end
